support/cas-server-support-radius/src/test/java/org/apereo/cas/config/RadiusConfigurationTests.java

```diff
@@ -0,0 +1,55 @@
+package org.apereo.cas.config;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apereo.cas.adaptors.radius.RadiusServer;
+import org.apereo.cas.configuration.CasConfigurationProperties;
+import org.apereo.cas.services.ServicesManager;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.cloud.autoconfigure.RefreshAutoConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.context.TestPropertySource;
+
+import java.util.List;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+/**
+ * @author Jozef Kotlar
+ * @since 5.3.0
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes={RadiusConfiguration.class, RefreshAutoConfiguration.class})
+@TestPropertySource(locations = {"classpath:/radius2.properties"})
+@Slf4j
+public class RadiusConfigurationTests {
+
+    @MockBean
+    @Qualifier("servicesManager")
+    private ServicesManager servicesManager;
+
+    @Autowired
+    private CasConfigurationProperties casProperties;
+
+    @Autowired
+    @Qualifier("radiusConfiguration")
+    private RadiusConfiguration radiusConfiguration;
```

Comment:
While this is excellent, let me also ask: would you be interested in putting together a few unit tests that demonstrate radius authentication functionality? You can try to spin a docker image that is based on free-radius and run tests against that. If so, let me know and I'll guide you through the process.

Reply:
Recommendations are implemented.

I cannot promise those tests in next weeks, but I will think about it meanwhile. Point me to some example.
I have actually freeradius server running for test in virtual - not docker, as on windows it is rather clumsy.

Reply:
Cool. Not a showstopper for this patch of course, but it would be great if you could contribute in that area when time permits.

The process is somewhat simple:

- When Travis CI starts, it will spin up a number of Docker images. You might need to find one based on freeradius that is configured with the essentials (i.e. protocol, client, secret, user accounts, etc). Then, you make sure Travis CI is able to start your container like it's done here for Couchbase:
https://github.com/apereo/cas/blob/master/.travis.yml#L50

- And you write a shell script that is able to start the freeradius docker container:
https://github.com/apereo/cas/blob/master/ci/run-couchbase-server.sh

- Finally, your test case will need to be just like any other test but it must be annotated with the following:

```java
@RunWith(ConditionalSpringRunner.class)
@ConditionalIgnore(condition = RunningContinuousIntegrationCondition.class)
public class MyTests {
...
}
```

This will make sure tests are only executed as part of Travis CI only. 

I should say that if you can find some sort of "embedded" free-radius server that can be started/stopped/configured programmatically, we can skip all the docker business and simply write normal tests...or just as well, if you can find a free cloud-based radiusserver that we can use a basic subscription for testing, that's fine too.